### PR TITLE
Use composite actions instead of a docker container

### PR DESCRIPTION
**Pull request by AndersBennedsgaard**
For an action such as this where a collection of tools are installed, it would make sense to write as a composite action.
This has several pros, such as:

* It allows to use the tools in other actions (this could be a solution to https://github.com/stefanprodan/kube-tools/issues/31)
* Much simpler setup and more DRY, which makes it easier to catch errors such as https://github.com/stefanprodan/kube-tools/issues/35
* Docker is not used, which may resolve issues such as https://github.com/stefanprodan/kube-tools/issues/36

It also has some cons, but the primary one is that no Docker containers are released. 
This could be solved, by simply using this action to deploy new container versions

I have tried to create a version which uses a composite action, but I did not want to change the entire setup of the action, without first discussing whether this is a favorable direction for the action :smiley:

### CODE REVIEW

**Comment by stefanprodan** [comment]
@AndersBennedsgaard this is really great. I wanted to switch this action to composite but I had no time to do it. What I think it would be great, is to allow setting `latest` as the version, if people don't want to pin a tool using semver. Since this is a major breaking change I will release it as v2.

**Comment by AndersBennedsgaard** [comment]
@stefanprodan that's good, I will get back to it in the next week. I agree that allowing the use of `latest`, would be a nice addition.
However, I don't think this is necessarily a breaking change, since to the user, it could behave exactly the same as before, but that is up to you.
With regards to the container-related things, should they persist in some form, or can they be removed?

**Comment by stefanprodan** [comment]
In v2 I would get rid of all the container-related things, no more `command`, also `helm` instead of `helmv3` as helmv2 has been deprecated long time ago.

**Comment by AndersBennedsgaard** [comment]
@stefanprodan I have tried to give a take on how to use the latest version, changed `helmv3` references to `helm`, and removed any Docker-related files (`Dockerfile`, `src/`, and `Makefile`). I am not the best at using the GitHub API, so some checks for the latest tag could probably be improved, and I am unsure on how the release workflow should behave now.

**Comment by stefanprodan** [comment]
@AndersBennedsgaard please remove Go from here, people should be using `actions/setup-go` directly and setup go.mod caching for it. Let's keep here only the statically build binaries.

**Comment by AndersBennedsgaard** [comment]
@stefanprodan I have removed Go, fixed the permissions error when installing `kubectl`, and added some caching which is not fully tested.

**Comment by stefanprodan** [comment]
CI is now failing with `/usr/local/bin/kustomize exists. Remove it first`.

**Comment by AndersBennedsgaard** [comment]
@stefanprodan I have removed `kustomize`. I did not realize how many tools come preinstalled with the GH runners, which is why my tests with [nektos/act](https://github.com/nektos/act) didn't catch the errors we saw

**Comment by stefanprodan** [comment]
@AndersBennedsgaard maybe we should try remove the binaries for all tools, you may never know when GitHub decides to add more things to the runner.

**Comment by AndersBennedsgaard** [comment]
@stefanprodan I've removed each tool in its respective install step, but I don't know if I agree that it's the responsibility of the action. In my opinion, the test workflow should include a check for whether the right version of the tool is installed

File: README.md [hunk 1]
```diff
@@ -4,7 +4,18 @@
...
+* kubectl
+* yq
+* kustomize
+* helm
+* kubeseal
+* kubeval
+* conftest
+* kubeaudit
+* kubeconform
+* kyverno
```
Comment (dangreenlee) [inline README.md:17]:
```suggestion
* [kubectl](https://github.com/kubernetes/kubectl)
* [yq](https://github.com/mikefarah/yq)
* [kustomize](https://github.com/kubernetes-sigs/kustomize)
* [helm](https://github.com/helm/helm)
* [kubeseal](https://github.com/bitnami-labs/sealed-secrets)
* [kubeval](https://github.com/instrumenta/kubeval)
* [conftest](https://github.com/open-policy-agent/conftest)
* [kubeaudit](https://github.com/Shopify/kubeaudit)
* [kubeconform](https://github.com/yannh/kubeconform)
* [kyverno](https://github.com/kyverno/kyverno)
```
Added links to the source repos. Helpful when selecting which version to input into the action.

**Review by dangreenlee — CHANGES_REQUESTED** [review]
Small change but might be helpful to a user of this action when selecting the version of a tool you want to install.

**Comment by dangreenlee** [comment]
> @stefanprodan I've removed each tool in its respective install step, but I don't know if I agree that it's the responsibility of the action. In my opinion, the test workflow should include a check for whether the right version of the tool is installed

If the test workflow checked to ensure the correct input version was installed, that would certainly better test the action as a whole, I think.

**Comment by AndersBennedsgaard** [comment]
> If the test workflow tested the input version and that the correct version was installed, that would certainly better test the action as a whole, I think.
> 
> Started to setup how this test may look here [AndersBennedsgaard#2](https://github.com/AndersBennedsgaard/kube-tools/pull/2).

@dangreenlee sorry I didn't get back to you sooner.
I didn't know about actions/github-release-info@v1 - I might include it in this content if @stefanprodan would prefer that change?
Unfortunately, not every tool uses uses the `latest` tag correctly, so the test will fail for tools such as `kubeseal`, `kubectl`, and `kustomize`. This is also why I've used the Github API for most of the installation steps.

**Pull request closed without merging**